test/e2e/csv_e2e_test.go

```diff
@@ -1320,6 +1320,380 @@ func TestCreateCSVWithOwnedAPIService(t *testing.T) {
+	require.NoError(t, err)
+
+	_, err = fetchCSV(t, crc, csv.Name, testNamespace, csvSucceededChecker)
+	require.NoError(t, err)
+
+	// Should create Deployment
+	_, err = c.GetDeployment(testNamespace, depName)
+	require.NoError(t, err, "error getting expected Deployment")
+
+	// Should create APIService
+	_, err = c.GetAPIService(apiServiceName)
+	require.NoError(t, err, "error getting expected APIService")
+
+	// Should create Service
+	_, err = c.GetService(testNamespace, olm.APIServiceNameToServiceName(apiServiceName))
+	require.NoError(t, err, "error getting expected Service")
+
+	// Should create certificate Secret
+	secretName := fmt.Sprintf("%s-cert", apiServiceName)
+	_, err = c.GetSecret(testNamespace, secretName)
+	require.NoError(t, err, "error getting expected Secret")
+
+	// Should create a Role for the Secret
+	_, err = c.GetRole(testNamespace, secretName)
+	require.NoError(t, err, "error getting expected Secret Role")
+
+	// Should create a RoleBinding for the Secret
+	_, err = c.GetRoleBinding(testNamespace, secretName)
+	require.NoError(t, err, "error getting exptected Secret RoleBinding")
+
+	// Should create a system:auth-delegator Cluster RoleBinding
+	_, err = c.GetClusterRoleBinding(fmt.Sprintf("%s-system:auth-delegator", apiServiceName))
+	require.NoError(t, err, "error getting expected system:auth-delegator ClusterRoleBinding")
+
+	// Should create an extension-apiserver-authentication-reader RoleBinding in kube-system
+	_, err = c.GetRoleBinding("kube-system", fmt.Sprintf("%s-auth-reader", apiServiceName))
+	require.NoError(t, err, "error getting expected extension-apiserver-authentication-reader RoleBinding")
+
+	// Create a new CSV that owns the same API Service and replace the old CSV
+	csv2 := v1alpha1.ClusterServiceVersion{
+		Spec: v1alpha1.ClusterServiceVersionSpec{
+			Replaces:       csv.Name,
+			MinKubeVersion: "0.0.0",
+			InstallModes: []v1alpha1.InstallMode{
+				{
+					Type:      v1alpha1.InstallModeTypeOwnNamespace,
+					Supported: true,
+				},
+				{
+					Type:      v1alpha1.InstallModeTypeSingleNamespace,
+					Supported: true,
+				},
+				{
+					Type:      v1alpha1.InstallModeTypeMultiNamespace,
+					Supported: true,
+				},
+				{
+					Type:      v1alpha1.InstallModeTypeAllNamespaces,
+					Supported: true,
+				},
+			},
+			InstallStrategy: v1alpha1.NamedInstallStrategy{
+				StrategyName:    install.InstallStrategyNameDeployment,
+				StrategySpecRaw: strategyRaw,
+			},
+			APIServiceDefinitions: v1alpha1.APIServiceDefinitions{
+				Owned: owned,
+			},
+		},
+	}
+	csv2.SetName("csv-hat-2")
+
+	// Create CSV2 to replace CSV
+	cleanupCSV2, err := createCSV(t, c, crc, csv2, testNamespace, false, true)
+	require.NoError(t, err)
+	defer cleanupCSV2()
+
+	_, err = fetchCSV(t, crc, csv2.Name, testNamespace, csvSucceededChecker)
+	require.NoError(t, err)
+
+	// Should create Deployment
+	_, err = c.GetDeployment(testNamespace, depName)
```

Comment:
All of this stuff will remain the same since the deployment didn't change. The only things that should get updated are the APIService CABundle and the generated Secret. But they should all still exist, so other than the comments here, this is probably a good check.